dev/release_notes.readme.md

```diff
@@ -0,0 +1,102 @@
+
+We show the subset of those merged pull requests that belong to the release in question and are regarded as relevant for the release notes.  For each of these pull requests, one line is shown, containing the number (as a link to the pull request in github) and the title.  Pull requests with related topics are grouped together under suitable subheadings.  These groups are ordered according to decreasing "severity".  Of course it is not possible to define such a general linear ordering, but we want each pull request to appear **only once** in the release notes, not in several groups according to several aspects it belongs to.
+
+Note that a file with release notes that is expected to be **read** must be **short**; other tools would be needed for **evaluating** the release notes (searching/filtering facilities according to various aspects).
+
+Besides the list of relevant pull requests, the release notes contain also the lists of changed GAP packages:
+- omitted packages (which had been distributed with the previous GAP version but are not distributed with the current one),
+- changed packages (which are distributed with both the previous and the current GAP version, but with different package versions),
+- new packages (which had not been distributed with the previous GAP version but are distributed with the current one).
+
+For each package in question, there is one entry that shows package name (with a link to the package homepage), version, authors and abstract.
+
+Note that the information about packages cannot be extracted from pull requests. It depends on the archives of packages that are distributed with GAP.
+(It would be easy to put this information together if there woud be a text file for each archive that lists which package versions are contained in the archive.)
+
+
+## How are release notes created from the github pull requests?
+
+The main idea is to use the *labels* that are assigned to the pull requests.
+Currently the following labels are regarded as relevant for the creation of release notes.
+
+- Pull requests with the label `release notes: not needed` shall **not** be listed in the release notes.
+  Thus the release notes are composed from all those merged pull requests that belong to the release in question (are assigned to the release branch or have a `backport-to-...-DONE` label for the release) AND do NOT have the label `release notes: not needed`.
+
+- For pull requests with the label `release notes: use title`, the text for the release notes is given by the title of the pull request.
+  For all other pull requests, the text for the release notes can be extracted from the pull request body.
```

Comment:
This is not done automatically yet. The easiest solution should be the following one at the moment:
If the description for the release notes is short enough, it should be the title and the PR should include the label "release notes: use title". So we do not use the "Text for release notes:" section here.
If a description is too long for the title, e.g. because too many different topics are addressed, there should be a label: "release notes: dont use title" such that the script gives us this PR for manual viewing.



Maybe we can try to expand the script at the next GAP Days such that we also get the text from PRs. We could then try to mark the release notes text for the script using certain start and end strings. E.g. :
ReleaseNotesScript_StartPoint
Text for the release notes......
ReleaseNotesScript_EndPoint